packages/contracts/contracts/rbac.clar

```diff
@@ -0,0 +1,169 @@
+(define-constant CONTRACT_OWNER tx-sender)
+
+;; Stores roles granted to users
+;; Each user can have up to 128 roles ranging from 0-127.
+;; Roles are stored as bitmask values represented as uint's.
+(define-map UserRoles
+  {user: principal}
+  {roles: uint}
+)
+
+;; Stores the roles granted to permissions
+;; Each permission can be granted to up to 128 roles ranging from 0-127
+;; Roles are stored as bitmask values represented as uint's.
+(define-map PermissionRoles
+  {permission: (string-ascii 50)}
```

Comment:
The roles should be defined as constants to avoid typos and make them available for off-chain operations.

Reply:
We don't know and will not know all permissions up front, therefore defining them in this contract as constants will not work.

What we can do is define permissions as constants per contract. RBAC contract will have only RBAC permissions, Vault contract will have only Vault permissions, SamplePlugin contract will have only SamplePlugin permissions etc.

Reply:
There is no way to return whole map all at once, so off-chain clients needs to pull values one by one.

We can introduce to RBAC contract a separate map and add to it new permission names. 
```
(define-map Permissions
  {id: uint}
  {name: (string-ascii 50)}
)
```
In addition to that expose `get-permissions-count` and `get-permission-name(id uint)` functions.

Reply:
I was only referring to the ones used in the contract here directly.

Reply:
OK, so we can and probably should define them as constants.
And use technique you pointed out [here](https://github.com/labs3/daoOS/pull/16#discussion_r620960336) to register them during contract deployment/initialization.

Reply:
Added constants to the code and have to admit that it looks much better.